game/game.cpp

```diff
@@ -24,6 +24,24 @@ namespace game
 
     int following = -1;
 
+    /**
+     * @brief The function takes a function or lambda and registers it as a
+     * CubeScript command.
+     * @tparam lambda a function or lambda.
+     * @param name the command name.
+     * @param f the function or lambda that will be bound to the command.
+     * @param args the number and type of arguments that the command takes, as
+     * described by CubeScript. See the definition of @ref addcommand if you are
+     * intersted in learning more about the types of arguments.
+     * @return false
+     */
+    template<typename lambda>
+    static constexpr bool addicommand(const char *name, lambda f, const char *args = "")
```

Comment:
A `template<class lambda>` will create a new function for every single time `addicommand<>` is called, because each lambda (even if it has the same signature) is assigned an anonymous type. I think that the use of `addcommand` on its own is sufficient here, and creating a template wrapper for `addcommand` creates a lot of single-use code upon compilation.

Reply:
What if functions are passed instead of lambdas? Would that prevent the functions from being assigned anonymous types?

Reply:
That would be the COMMAND macro, which takes a global function's name as a parameter. ICOMMAND, or Inline COMMAND, is for declaring the execution body inside the macro declaration.